framework/aster-frame/src/task/processor.rs

```diff
@@ -56,6 +59,9 @@ pub(crate) fn get_idle_task_cx_ptr() -> *mut TaskContext {
 
 /// call this function to switch to other task by using GLOBAL_SCHEDULER
 pub fn schedule() {
+    unsafe {
```

Comment:
Every usage of `unsafe` should be justified with well-written comment. Inserting `pass_quiescent_state` is actually  unsound. Consider the following code.

```rust
let read_guard = rcu.get();
use_rcu_data(read_guard);
schedule(); // Oops: RCU-protected data is accessed across a quiescent state!
use_rcu_data(read_guard);
```

To prevent such situations, `pass_quiescent_state` must be called _after_ checking whether any preemption guards exist.